File: Courzelo-backend-LMS/Courzelo/src/main/java/org/example/courzelo/models/RevisionEntities/QizzRevision/QuizRevision.java
package org.example.courzelo.models.RevisionEntities.QizzRevision;


import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.example.courzelo.models.RevisionEntities.revision.FileMetadatarevision;
import org.example.courzelo.models.RevisionEntities.revision.Revision;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Document(collection = "QuizRevision")
public class QuizRevision {
    @Id
    private String id;

    @Indexed
    private String title;

    @DBRef
    private FileMetadatarevision fileMetadatarevision;

    @DBRef
    private List<QuestionRevision> questions;

    @DBRef
    private Revision revision;
}

File: courzelo-dashboard-LMS-main/src/app/shared/models/Revision/QuizzRevisison/QuizRevision .ts
import { FileMetadatarevision } from "../FileMetadatarevision";
import { QuestionRevision } from "./QuestionRevision";
import { Revision } from "../Revision";

export interface QuizRevision {
    id: string;
    title: string;
    fileMetadatarevision: FileMetadatarevision;
    questions: QuestionRevision[];
    revision: Revision;
  }